### PR TITLE
to and from fields added

### PR DESCRIPTION
**Pull request by derins**
# Current issues
none

# Code changes
1. User input for file type from and file type to is added
2. config.py file implemented instead of app.config calls in app.py
3. got rid of main.py for app.py

# Documentation changes
none

# Validation suggestions
1. Set up the environment using commands in README.md
2. run 
```
python app.py
```
3. go to the address that comes up (it will likely be "localhost:5000" on browser)
4. go to convert tab and upload either of the json files in test_files

### CODE REVIEW

**Comment by dchassin** [comment]
One caveat: GridLAB-D does not support running python in the environments like `conda`.  This means that although running standalone is `conda` is ok, the converters themselves and the converter configuration API used to identify supported converters and options must not depend on anything that doesn't work in the generic python environment.

**Comment by dchassin** [comment]
Could we not make any changes to the converters copied from the GridLAB-D project until the converter API is available and working?  There is no way for GridLAB-D to call these as they are now.  The calling convention from the command line is standardized and needs to remain until then.  The calling convention is `python3 <ext>2<ext>.py -i <input> -o <ouput> [-t <type>] [options]`. As it stands now, there's no way for GridLAB-D to call either of the converters listed, nor is there any way for GridLAB-D to identify which file formats and types are available.

We can/should add a config dict and an entry function so that converters can be called directly in python if possible in the calling app (e.g., GridLAB-D cannot be run in conda, so it uses `system` calls):

~~~
>>> import json2glm
>>> print(json2glm.config)
{"input":"json","output":"glm","type":[]}
>>> json2glm.convert(input=<file>,output=<file>,type=<file>)
~~~

This means that the CLI code needs to be changed so that import does not attempt to run the CLI parsers, but command line calls still work, e.g.,:

~~~
bash$ python3 json2glm -c
{"input":"json","output":"glm","type":[]}
bash$ python3 json2glm -i myfile.json -o myfile.glm
~~~

Also, the use of an `uploads` folder seems related to the app, not a converter itself.  It should not be part of how the converters themselves work.

**Comment by jongoncalves** [comment]
> One caveat: GridLAB-D does not support running python in the environments like `conda`. This means that although running standalone is `conda` is ok, the converters themselves and the converter configuration API used to identify supported converters and options must not depend on anything that doesn't work in the generic python environment.

Point well taken, however, irrelevant. The goal, again, is not for GLD to use this directly; this project, aside from demo purposes (and web integration to the OpenFIDO/CMU work), serves as an intermediary step to transfer these converters into a package (i.e. pypi or anaconda cloud).

Ultimately GLD will be able to call the converter from the package via the cmd line like:
`python convert.py -i <if> -o <of>.....`

File: app.py [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
+import json
+import sys
+import getopt
+import shutil
+import glob
+from flask import Flask, flash, request, redirect, render_template, url_for, send_from_directory
+from werkzeug.utils import secure_filename
 
-UPLOAD_FOLDER = './uploads'
+ALLOWED_EXTENSIONS = set(['json'])
```
Comment (jongoncalves) [inline app.py]: This param should be in the settings file.

File: app.py [hunk 2]
```diff
@@ -1,17 +1,65 @@
...
 
 app = Flask(__name__)
-app.secret_key = "secret key"
-app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
-app.config['MAX_CONTENT_LENGTH'] = 16 * 1024 * 1024
+app.config.from_object("settings.config.DevelopmentConfig")
 
+def allowed_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
```
Comment (jongoncalves) [inline app.py]: I know this is straight out of the Flask docs, but it's better to use the builtin helpers to do this, i.e.:
```python
import os

name, extension = os.path.splitext(filename)
extension = extension.lower()

extension in ALLOWED_EXTENSIONS
...
```

File: app.py [hunk 3]
```diff
@@ -1,17 +1,65 @@
...
 
 @app.route('/')
 def index():
     return render_template('home.html')
 
+@app.route('/convert')
+def upload_form():
+    # content = log()
```
Comment (jongoncalves) [inline app.py]: remove comment

File: app.py [hunk 4]
```diff
@@ -1,17 +1,65 @@
...
+    return render_template('convert.html')
+
+@app.route('/', methods=['POST'])
+def upload_file():
+    if request.method == 'POST':
```
Comment (jongoncalves) [inline app.py]: You specified `POST` as the supported method in the decorator, you shouldn't need to check `request.method`. Please verify.

File: app.py [hunk 5]
```diff
@@ -1,17 +1,65 @@
...
+        convertFrom = request.form["convertFrom"]
+        convertTo = request.form["convertTo"]
+        uploadFiles = glob.glob('./uploads/*')
```
Comment (jongoncalves) [inline app.py]: Though the naming convention coming from JS (i.e. camelCase) is appropriate, please use the correct convention in python, snake_case, for all your variable names

File: static/js/convert.js [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (jongoncalves) [inline static/js/convert.js:1]: What's this blank file for?

File: static/js/display_output.txt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+ERROR    [INIT] : strict naming prevents implicit creation of powerflow::solver_ml_config
+../test_files/test_IEEE_13_NR.glm(50): invalid module powerflow property 'solver_ml_config'
+../test_files/test_IEEE_13_NR.glm(50): expected module powerflow block closing }
+../test_files/test_IEEE_13_NR.glm(9): load failed at or near 'module power...'
+FATAL    [INIT] : shutdown after command line rejected
```
Comment (jongoncalves) [inline static/js/display_output.txt]: This file should be removed/gitignored

File: uploads/test_lights.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+ERROR: json to txt conversion not implemented yet
```
Comment (jongoncalves) [inline uploads/test_lights.txt]: Same as before. Why is this file here? It should likely be removed or gitignored.

File: settings/config.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    pass
+
+
+class DevelopmentConfig(Config):
+    DEBUG = True
+    UPLOAD_FOLDER = "./uploads"
+
+
+class TestingConfig(Config):
+    TESTING = True
```
Comment (jongoncalves) [inline settings/config.py]: What's this TESTING param used for? I realize it's in the Flask docs, but it doesn't seem like we are using it, so remove it.

**Review by jongoncalves — COMMENTED** [review]
Why was the `environment.yml` removed?
Also, please update the README.md accordingly and add in test execution instruction.

File: app.py (hunk 1 above)
Comment (derins) [reply app.py]: Done

File: app.py (hunk 2 above)
Comment (derins) [reply app.py]: Done

File: app.py (hunk 3 above)
Comment (derins) [reply app.py]: Done

File: app.py (hunk 4 above)
Comment (derins) [reply app.py]: You are right, I removed that if statement

File: app.py (hunk 5 above)
Comment (derins) [reply app.py]: Fixed

File: static/js/convert.js (hunk 6 above)
Comment (derins) [reply static/js/convert.js:1]: I removed some redundant javascript functions from that file, I will write some functions to filter dropdown options.

File: static/js/display_output.txt (hunk 7 above)
Comment (derins) [reply static/js/display_output.txt]: done

File: uploads/test_lights.txt (hunk 8 above)
Comment (derins) [reply uploads/test_lights.txt]: removed the file

File: settings/config.py (hunk 9 above)
Comment (derins) [reply settings/config.py]: removed it

File: app.py [hunk 10]
```diff
@@ -1,17 +1,62 @@
...
+import json
+import sys
+import getopt
+import shutil
+import glob
+from flask import Flask, flash, request, redirect, render_template, url_for, send_from_directory
+from werkzeug.utils import secure_filename
 
-UPLOAD_FOLDER = './uploads'
+# ALLOWED_EXTENSIONS = set(['json'])
```
Comment (jongoncalves) [inline app.py]: remove comment

File: app.py [hunk 11]
```diff
@@ -1,17 +1,62 @@
...
+        flash('No file part')
+        return redirect(request.url)
+    file = request.files['file']
+    if file.filename == '':
+        flash('No file selected for uploading')
+        return redirect(request.url)
+    if file and allowed_file(file.filename):
+        filename = secure_filename(file.filename)
+        file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+        os.system('python ./' + convert_from + '2' + convert_to + '.py')
```
Comment (jongoncalves) [inline app.py:45]: This is ok for now, but the next PR should address this. We shouldn't be calling the OS to call python to run a script. Make converter a decoupled, package-able component that gets called in code via a simple function call.

File: app.py (hunk 10 above)
Comment (derins) [reply app.py]: Done

File: app.py (hunk 11 above)
Comment (derins) [reply app.py:45]: Sounds good

**Review by jongoncalves — APPROVED** [review]
